topic_dynamics/parsing.py

```diff
@@ -172,42 +163,78 @@ def slice_and_parse(repository: str, output_dir: str, dates: List, lang: str, na
     with open(os.path.abspath(os.path.join(output_dir, name + "_slices.txt")), "w+") as fout:
         for date in dates_indices.keys():
             if dates_indices[date][1] >= dates_indices[date][0]:  # Skips empty slices
-                fout.write(date + ";" + str(dates_indices[date][0]) + "," + str(dates_indices[date][1]) + "\n")
+                fout.write(";".join([date, str(dates_indices[date][0]), str(dates_indices[date][1])]) + "\n")
 
 
 def split_token_file(slices_file: str, tokens_file: str, output_dir: str) -> None:
     """
-    Split a single file with tokens into splits for calculating diffs.
+    Split a single temporary file with tokens into splits for calculating diffs. A single
+    tokens file with files indices (1, 2, 3, 4, 5, 6) is taken and split into separate files
+    1: (1, 2), 2: (3, 4), 3: (5, 6) for simpler processing while calculating diffs.
     :param slices_file: the address of the file with the indices of the slices.
     :param tokens_file: the address of the temporary file with tokens.
     :param output_dir: path to the output directory.
     :return: None.
     """
     print("Splitting the tokens of full files by versions.")
     slice_number = 0
-    dates_indices = {}
+    date2indices = {}
     if not os.path.exists(output_dir):
         os.makedirs(output_dir)
     # Read the data about the indices boundaries of slices
     with open(slices_file) as fin:
         for line in fin:
             slice_number = slice_number + 1
-            dates_indices[slice_number] = (int(line.rstrip().split(";")[1].split(",")[0]),
-                                           int(line.rstrip().split(";")[1].split(",")[1]))
+            slice_line = SliceLine(*line.split(";"))
+            date2indices[slice_number] = (int(slice_line.starting_index),
```

Comment:
conversion to correct type should be done during initialization - [example](https://stackoverflow.com/questions/44614504/type-conversion-for-namedtuple-fields-during-initialization)